File: 01-RAG智能客服系统/backend/app/schemas/kb.py
"""Pydantic schemas for knowledge base management."""
from datetime import datetime
from pydantic import BaseModel, Field


class DocumentResponse(BaseModel):
    """Document information."""
    id: str
    filename: str
    original_name: str
    file_type: str
    file_size: int
    chunk_count: int
    char_count: int
    status: str
    error_message: str | None = None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class DocumentListResponse(BaseModel):
    """Paginated document list."""
    items: list[DocumentResponse]
    total: int
    page: int
    page_size: int


class KBStatsResponse(BaseModel):
    """Knowledge base statistics."""
    doc_count: int
    chunk_count: int
    total_chars: int
    total_size_bytes: int
    last_updated: datetime | None = None


class JobStatusResponse(BaseModel):
    """Document processing job status."""
    job_id: str
    status: str  # pending | processing | completed | failed
    progress_pct: float = 0.0
    processed: int = 0
    total: int = 0
    errors: list[str] = []

File: 07-MCP多智能体协作平台/frontend/src/api/kb.ts
import apiClient from './client'

export interface Document {
  id: string
  filename: string
  original_name: string
  file_type: string
  file_size: number
  chunk_count: number
  char_count: number
  status: 'pending' | 'processing' | 'completed' | 'failed'
  error_message: string | null
  created_at: string
  updated_at: string
}

export interface KBStats {
  doc_count: number
  chunk_count: number
  total_chars: number
  total_size_bytes: number
  last_updated: string | null
}

export const kbApi = {
  listDocuments(page = 1, pageSize = 20, status?: string) {
    return apiClient.get<{ items: Document[]; total: number }>('/kb/documents', {
      params: { page, page_size: pageSize, status },
    })
  },
  uploadDocument(file: File) {
    const formData = new FormData()
    formData.append('file', file)
    return apiClient.post<Document>('/kb/documents/upload', formData, {
      headers: { 'Content-Type': 'multipart/form-data' },
    })
  },
  deleteDocument(docId: string) {
    return apiClient.delete(`/kb/documents/${docId}`)
  },
  reprocessDocument(docId: string) {
    return apiClient.post(`/kb/documents/${docId}/reprocess`)
  },
  getStats() {
    return apiClient.get<KBStats>('/kb/stats')
  },
}
